viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
+            genre_list.append(movie["genre"])
+        return max(set(genre_list), key = genre_list.count)
 
+        
+
+
+
+# -----------------------------------------
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    movies_list = [movies for movies in user_data["watched"]]
+    for movie in user_data["friends"]:
+        for title in movie["watched"]:
+            if title in movies_list:
+                movies_list.remove(title)
+    return movies_list
+            
+def get_friends_unique_watched(user_data):
+    movies_list = [movies for movies in user_data["watched"]]
+    friends_list = []
+    for movie in user_data["friends"]:
+        for title in movie["watched"]:
+            if title not in movies_list:
+                if title not in friends_list:
+                    friends_list.append(title)
+    
+    return friends_list
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    friends_recs = []
+    compatible = [] 
+    movies_list = [movies for movies in user_data["watched"]]
+    for movie in user_data["friends"]:
+        for title in movie["watched"]:
+            if title not in movies_list:
+                if title not in friends_recs:
+                    friends_recs.append(title)
 
+    for movie in friends_recs:
+        if movie["host"] in user_data["subscriptions"]:
+            compatible.append(movie)
+    return compatible
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    genre_recs = []
+    genre_list = []
+    fave_genre = []
+   
+    for movie in user_data["watched"]:
+        if len(user_data["watched"]) == 0:
+            return None
+        genre_list.append(movie["genre"])
+        
+
+        fave_genre.append(max(set(genre_list), key = genre_list.count))
+
+    friends_recs = []
+    movies_list = [movies for movies in user_data["watched"]]
+    for movie in user_data["friends"]:
+        for title in movie["watched"]:
+            if title not in movies_list:
+                if title not in friends_recs:
+                    friends_recs.append(title)
+    for movie in friends_recs:
+        if movie["genre"] in fave_genre:
+            genre_recs.append(movie)
+
+    return genre_recs
+
+def get_rec_from_favorites(user_data):
+    movie_recs = []
+
+    movies_list = [movies for movies in user_data["favorites"]]
+    friends_list = []
+    for movie in user_data["friends"]:
+        for title in movie["watched"]:
+            if title not in friends_list:
+                friends_list.append(title)
+    for movie in movies_list:
+        if movie not in friends_list:
+            # if movie not in movie_recs:

```

Comment:
Remember to remove unused commented out code to keep your code clutter free. Also it enables us to avoid a scenario where the code gets accidentally _uncommented_ and this line of code that you never wanted to execute ends up executing